[Feature] Retire government employee section

yonikid15 opened the pull request:
🤖 Resolves #12140 

## 👋 Introduction

- Adds new priority entitlements section to profile page and application profile step
- Changes government information section to read-only
- Doesn't make any changes to the admin side or downloads (should be handled in #15045)

## 🕵️ Details

The issue asks to remove fields from their mutations that are no longer needed, but after checking again I wasn't sure which inputs no longer need them.

- CreateUserInput
- UpdateUserAsAdminInput 
- UpdateUserAsUserInput **(Update: removed from here)**

## 🧪 Testing

1. Build app
2. Login as anyone
3. Go to profile information page
4. Ensure everything matches the design and works correctly
5. Create an application and go to the second step (review profile)
6. Ensure everything matches the design and works correctly

## 📸 Screenshot

<img width="3046" height="1455" alt="image" src="https://github.com/user-attachments/assets/37e2895d-e92a-416a-a558-8ee051457114" />

<img width="3046" height="1455" alt="image" src="https://github.com/user-attachments/assets/d8245307-0234-4353-8cbc-909a10dacfff" />

yonikid15 commented:
> Hi Yoni 😁 I noticed this has changed how government employee information appears in the admin user table view
>

Nice catch @bella77-69 ! Should be fixed in [Fix admin view](https://github.com/GCTC-NTGC/gc-digital-talent/pull/15137/commits/77a28f9e0989a6108060e345b1718f4ebd490a63)

brindasasi reviewed (changes requested):
Confirmed #2 exists in main as well.. will create a  sep issue for that  .. but the #1 should be fixed in under this issue

yonikid15 commented:
> Hey Yoni, Sorry I find one more issue on this flow.
> 
> 1. Work experience section in Personal Information page doesn't get refreshed with latest values when you navigate to that.
>    It only updates when you click refresh
> 
> Here in the video , I update a gov exp to non-gov exp and navigate back to Personal Information , still holds the old value until I click page refresh explicitly
> 

Okay I changed the request policy to cache and network which should now fetch the updated data. Changed in [Change request policy to cache and network](https://github.com/GCTC-NTGC/gc-digital-talent/pull/15137/commits/df6ea8a88664d3d294db43ec972e7097cba3c286)

brindasasi reviewed (approved):
LGTM!
